src/main/java/jenkins/plugins/logstash/persistence/BuildData.java

```diff
@@ -256,19 +256,27 @@ private void initData(Run<?, ?> build, Date currentTime) {
     description = build.getDescription();
     url = build.getUrl();
     buildNum = build.getNumber();
+    buildDuration = currentTime.getTime() - build.getStartTimeInMillis();
+    timestamp = DATE_FORMATTER.format(build.getTimestamp().getTime());
+  }
 
+  private void updateResult()
```

Comment:
why not move this to `getTestResults()`?

Reply:
getTestResults() is just a getter. And we also have getResult(). The updateResult() is updating both.
And honestly BuildData should not really be public api and used outside of logstash. This is an implementation detail.

Reply:
OK, can we do something else to remove the need for a explicit call to `buildData.toJson();` in  tests?